forte/pipeline.py

```diff
@@ -298,6 +300,7 @@ def init_from_config(self, configs: Dict[str, Any]):
                         class_name=selector_config["type"],
                         class_args=selector_config.get("kwargs", {}),
                     ),
+                    selector_config=selector_config.get("configs")
```

Comment:
There might be cases where `component_config` does not have a `"selector"` field. If that's the case, maybe we can assume both `selector` and `selector_config` take the default value `None`.

Reply:
We do have `__default_selector` and `__default_selector_config` that will be assigned if there is no `selector`

https://github.com/zhanyuanucb/forte/blob/ded754c482bb897a0fb4b918d0680df4e5441415/forte/pipeline.py#L817-L819

Reply:
I mean the the variable `selector_config` here might be `None`, so calling `selector_config.get()` may raise an exception.

Reply:
Do you mean we should double-check whether `component_config.get("selector")` returns a `None`?

hmmm... I don't think `select_config` will ever be `None` given that every processor is coupled with at least a `__default_selector` based on what happens in the `add` method.

Reply:
Yes, `selector_config` will never be `None` for any config generated after this PR is merged. Adding this check is only meant to make it easier to fix some old config file like https://github.com/asyml/forte-wrappers/blob/main/examples/gpt2/sample_multipack_pipeline_gpt.yml. Anyway, we can leave it as it is.

Reply:
I see your point.
I also noticed that there is an IR version validation in `init_from_config`:https://github.com/zhanyuanucb/forte/blob/init_selector/forte/pipeline.py#L269-L275

We could rely on this validation to indirectly ask users to switch to a Forte version that is compatible to their Forte IR, but the `inig_from_config` in this PR won't be backward compatible.